File: 1001-n-repeated-element-in-size-2n-array/1001-n-repeated-element-in-size-2n-array.java
class Solution {
    public int repeatedNTimes(int[] nums) {
        int num_l = nums.length;
        int n = (num_l >> 1);
        int[] hash = new int[10001];
        for (int i = 0; i < num_l; ++i) {
            hash[nums[i]]++;
            if (hash[nums[i]] >= n)
                return nums[i];
        }
        return -1;
    }
}

File: 1001-n-repeated-element-in-size-2n-array/1001-n-repeated-element-in-size-2n-array.go
func repeatedNTimes(nums []int) int {
    length := len(nums)
    hash := make([]bool, 10001)

    for i := 0; i < length; i++ {
        if hash[nums[i]] == true {
            return nums[i]
        } else {
            hash[nums[i]] = true
        }
    }
    return -1
}